src/test/java/net/openhft/compiler/CompilerTest.java

```diff
@@ -240,10 +268,10 @@ public void write(int b) throws IOException {
                 }
             }));
 
-            CompilerUtils.CACHED_COMPILER.loadFromJava(
-                    getClass().getClassLoader(), "TestClass",
-                    // definition with a mandatory warning
-                    "class TestClass { int i = new Date().getDay(); }",
+            compiler.loadFromJava(
+                    classLoader, "TestClass",
+                    // definition with a mandatory warning for deprecated `Date.getDay()`
+                    "import java.util.Date; class TestClass { int i = new Date().getDay(); }",
```

Comment:
In this line there is one of the reasons why I changed this test class to create a new `CachedCompiler` instance for every test method, instead of using `CompilerUtils.CACHED_COMPILER` and `getClass().getClassLoader()`:
It seems this source code was missing the `import java.util.Date`, which should normally cause a compiler error. However, previously no compilation error occurred, probably because one of the other test methods had already successfully loaded `TestClass` and therefore the `CachedCompiler#loadFromJava` call just returned that class instead of compiling the code.